tests/tests/regression_tests.rs

```diff
@@ -1631,3 +730,87 @@ impl ExternalInterfaceProvider for ExternalInterfaceTestProvider {
         false
     }
 }
+
+fn run_test(options: TestOptions, root: &Path) -> Result<(), libtest_mimic::Failed> {
+    set_logger();
+
+    if let Some(approximations) = &options.approximations {
+        let num_patterns: Vec<Regex> = approximations
+            .number_patterns
+            .iter()
+            .map(|p| Regex::new(&p).unwrap())
+            .collect();
+        test_swf_approx(
+            root.join("test.swf").to_str().unwrap(),
+            options.num_frames,
+            root.join("input.json").to_str().unwrap(),
+            root.join("output.txt").to_str().unwrap(),
+            &num_patterns,
+            options.image,
+            |actual, expected| approximations.compare(actual, expected),
+        )
+        .map_err(|e| e.to_string().into())
+    } else {
+        test_swf_with_hooks(
+            root.join("test.swf").to_str().unwrap(),
+            options.num_frames,
+            root.join("input.json").to_str().unwrap(),
+            root.join("output.txt").to_str().unwrap(),
+            |player| {
+                if let Some(player_options) = &options.player_options {
+                    player_options.setup(player);
+                }
+                Ok(())
+            },
+            |_| Ok(()),
+            options.image,
+            options.sleep_to_meet_frame_rate,
+        )
+        .map_err(|e| e.to_string().into())
+    }
+}
+
+fn main() {
+    let args = Arguments::from_args();
+
+    let root = Path::new("tests/swfs");
+    let mut tests: Vec<Trial> = walkdir::WalkDir::new(root)
+        .into_iter()
+        .map(Result::unwrap)
+        .filter(|entry| entry.file_type().is_file() && entry.file_name() == "test.toml")
+        .map(|file| {
+            let options: TestOptions =
+                toml::from_str(&fs::read_to_string(&file.path()).unwrap()).unwrap();
+            let test_dir = file.path().parent().unwrap().to_owned();
+            let name = test_dir
+                .strip_prefix(root)
+                .unwrap()
+                .to_string_lossy()
+                .replace('\\', "/");
+            let ignore = options.ignore || (options.image && !RUN_IMG_TESTS);
```

Comment:
Do we have any tests currently that would be `needs_render_backend` but not `image`, or is this for future tests?

Reply:
At the moment, I think we just have:

*  `image` without `needs_render_backend` (test that just modify the display tree and want to check the final output)
* `image` with `needs_render_backend` (Stage3d tests that need to draw things with a context3d, but also compare the final output).
*
In the future, we could have a test that does BitmapData.draw and just manually compares the pixel values (maybe some kind of performance test). That would be `needs_render_backend` but not `image`

Reply:
I'm confused, any image needs a render background because how else will it get the image? Image required render.

Since this deals with a future case, can we discuss it on slack and leave it for a future PR? I'd like to change how image tests work (add tolerances) and that may be a change we can bundle with it